projects/core/directives/dropdown/dropdown-context.directive.ts

```diff
@@ -37,8 +41,8 @@ export class TuiDropdownContextDirective extends TuiDriver implements TuiRectAcc
     @HostListener(`document:click.silent`, [`$event.target`])
     @HostListener(`document:contextmenu.capture.silent`, [`$event.target`])
     @HostListener(`document:keydown.esc`, [`$event.currentTarget`])
-    closeDropdown(target: EventTarget): void {
-        if (!this.dropdown.dropdownBoxRef?.location.nativeElement.contains(target)) {
+    closeDropdown(target: Element): void {
+        if (!this.activeZone.contains(target)) {
```

Comment:
Is it possible to add e2e for check this case?

Reply:
Done